### PR TITLE
Improve proxy support for install scripts

### PR DESCRIPTION
**Pull request by rthorn-nr**
For `install.sh` and `install.ps1`, check for connectivity to `https://download.newrelic.com` and provide a helpful message if unable to connect.

For `install.ps1`, adds a proxy to the requesting `WebClient` via the optional environment variable `HTTPS_PROXY`, which will enable setting a proxy for the CLI download.

### CODE REVIEW

File: scripts/install.ps1
Comment (Julien4218) [inline scripts/install.ps1:15]: It might be good to get the exception, and add the inner message below.

Comment (Julien4218) [reply scripts/install.ps1:15]: We should test this as I recall some weirdness sometimes with powershell and try/catch not catching all exceptions.

Comment (Julien4218) [reply scripts/install.ps1:15]: Is there a way to put a connect timeout? I don't think it should be more than 10s, just like the linux shell version.

File: scripts/install.ps1
Comment (Julien4218) [inline scripts/install.ps1:8]: Is there a reason why we wouldn't use the same HTTPS_PROXY mechanism for windows?

**Review by Julien4218 — COMMENTED** [review]
@rthorn-nr I left a few comments.

File: scripts/install.ps1
Comment (rthorn-nr) [reply scripts/install.ps1:8]: No specific reason other than thinking that it would be more flexible to pass it as a script parameter than to set the user environment, but also no reason to break from convention. Pushing up a change for this momentarily

File: scripts/install.ps1
Comment (rthorn-nr) [reply scripts/install.ps1:14]: > Is there a way to put a connect timeout?

At first glance, it doesn't look like there's a way to do this with [WebClient](https://learn.microsoft.com/en-us/dotnet/api/system.net.webclient?view=net-7.0) (unlike the newer [HttpClient](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient?view=net-7.0), which has a timeout property). There's a [workaround](https://stackoverflow.com/a/12879118) which involves extending the class, but I'm seeing a timeout after about 20s... if we want to get that down to 10s, it'll just take some reworking one way or another, so I'd leave it up to you whether we want to shorten the feedback loop.

Comment (rthorn-nr) [reply scripts/install.ps1:14]: Going to do some testing with exceptions and get back to you on those comments

Comment (Julien4218) [reply scripts/install.ps1:14]: I was going to suggest to keep what we have in regard of the timeout, but comparing the requirements for each of the method (apply to at the bottom), I noticed the HttpClient has broader support than WebClient. Also the doc on WebClient mentions "We don't recommend that you use the WebClient class for new development. Instead, use the [System.Net.Http.HttpClient](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient?view=net-7.0) class."
I recall using the HttpClient in the past, I believe it's a more bare bone way to make request. It might be good to switch to using it actually, and avoid compatibility issues for hosts that don't have the right .NET version.

Comment (rthorn-nr) [reply scripts/install.ps1:15]: Error output should be improved by the latest [commit](https://github.com/newrelic/newrelic-cli/pull/1432/commits/766fef757378d0f11471f76f9382ab3d2a42ce26). Moved the file download to the try/catch as well. 

I'd naively assume that a regular try/catch would catch all exceptions coming out of those two download commands -- let me know how I can help test that powershell try/catch weirdness

Comment (rthorn-nr) [reply scripts/install.ps1:15]: Missed your comment above yesterday. Makes sense, happy to refactor to use HttpClient

**Review by Julien4218 — DISMISSED** [review]
Approving after verifying on linux and windows with proxy.

File: scripts/install.ps1
Comment (rthorn-nr) [reply scripts/install.ps1:15]: @Julien4218 I think we should stick with WebClient for now. 

Testing using PS v5.1 (default for Server 2019 and Server 2022), HttpClient's async behavior has proven hard to handle in a PS script. Using a strategy like [this](https://stackoverflow.com/a/51218694) and calling `GetResult()` off of `GetAwaiter()` cancels the request, and the `await` syntax is not available. 

I also tried using `Invoke-WebRequest`, but that triggered an IE security warning: `Content within this application coming from the website listed below is being blocked by Internet Explorer Enhanced Security Configuration: about:security_powershell.exe`. Might just by my testing setup, but using WebClient does not trigger this warning.

It's a shame, especially since HttpClient automatically reads proxy variables from the environment, as does `Invoke-Webrequest` in PS versions 7+. Seems to me like it's still a bit of a compatibility risk to update this script in that way.

**Review by Julien4218 — APPROVED** [review]

**Pull request merged**
